### ISSUE TITLE
$! is set to EBADF when it should be ENOENT

### ISSUE THREAD

**Issue by FGasper** [cpanel/Test-MockFile#55]
```perl
> perl -MTest::MockFile -e'my $file = Test::MockFile->file("/foo/bar", undef); 
die "huh" if -e "/foo/bar"; print "$!"'
Bad file descriptor
```
^^ The above should say something like, `No such file or directory`.

This breaks anything that verifies $! after `-e` fails (which probably ought to be _almost_ everything).

**Comment by xsawyerx** [comment]
I've provided a preliminary PR to handle it here: https://github.com/cpanel/Overload-FileCheck/pull/20

Essentially, because `Overload::FileCheck` sets the cache and then calls `stat`, it triggers the path in perl that returns `EBADF` instead of calling `stat` (which would have returned `ENOENT`).

**Comment by xsawyerx** [comment]
This is now fixed with [Overload::FileCheck 0.011](https://metacpan.org/release/ATOOMIC/Overload-FileCheck-0.011/view/lib/Overload/FileCheck.pm).

```
$ perl -Ilib -MTest::MockFile -E'my $file = Test::MockFile->file("/foo/bar", undef); die "huh" if -e "/foo/bar"; print "$!"' 
No such file or directory
```
